packages/x-date-pickers-pro/src/internal/hooks/useMultiInputRangeField/useMultiInputDateRangeField.ts

```diff
@@ -37,6 +37,7 @@ export const useMultiInputDateRangeField = <TDate, TChildProps extends {}>({
   const adapter = useLocalizationContext<TDate>();
 
   const { value: valueProp, defaultValue, format, onChange, disabled, readOnly } = sharedProps;
+  const { shouldControlSelection, ...otherSharedProps } = sharedProps;
```

Comment:
Why should it be extracted from `sharedProps` if it is shared between the two inputs?